src/Orchestration/Adapter/DockerAPI.php

```diff
@@ -319,9 +319,9 @@ public function run(string $image, string $name, array $command, string $entrypo
      */
     public function execute(string $name, array $command, string &$stdout, string &$stderr, array $vars = [], int $timeout = -1): bool
     {
-        foreach ($vars as $key => &$value) {
+        foreach ($vars as $key => $value) {
             $key = $this->filterEnvKey($key);
-            $value = $key.'='.$value;
+            $vars[$key] = $key.'='.$value;
```

Comment:
This might create duplicates if the filtered `$key` is not the same as the non filtered one.